code-challenges.test.js

```diff
@@ -11,42 +11,174 @@
 // Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
+const secretCodeWord2 = "Gobbledygook";
 // Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
+const secretCodeWord3 = "Eccentric";
 // Expected output: "3cc3ntr1c"
 
-// b) Create the function that makes the test pass.
+const codedMessage = (string) => {
+  for (let i = 0; i < string.length; i++) {
+    const lowerCaseString = string.toLowerCase();
+    {
+      if (lowerCaseString.includes("a", "e", "i", "o")) set: a = "4";
+      set: e = "3";
+      set: i = "1";
+      set: o = "0";
+    }
+  }
+  return lowerCaseString;
+};
+
+console.log(codedMessage(secretCodeWord1));
+console.log(codedMessage(secretCodeWord2));
+console.log(codedMessage(secretCodeWord3));
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
+// psuedocode:
+// create function named filterLetter that takes in a array and outputs an array function must look at each letter individually so i may create an emplty array and push the given variables into it. iterate over the array lookinng at each value in each index filtering or searching for the specific letter
+// input: array and "string"
+// output:"string"
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const letterA = "a"
+describe("filterLetter", () => {
+  it("takes in an array of words and a single letter and returns an array of all the words containing that particular letter.", () => {
+    const fruitArray = [
+      "Mango",
+      "Cherry",
+      "Apricot",
+      "Blueberry",
+      "Peach",
+      "Kiwi",
+    ];
+
+    const letterA = "a";
+    // Expected output: ["Mango", "Apricot", "Peach"]
+    const letterE = "e";
+    // Expected output: ["Cherry", "Blueberry","Peach"]
+
+    expect(filterLetter(fruitArray(lettterA))).toEqual([
+      "Mango",
+      "Apricot",
+      "Peach",
+    ]);
+    expect(filterLetter(fruitArray(letterE))).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+  });
+});
+
+const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"];
+
+const letterA = "a";
 // Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+const letterE = "e";
+// Expected output: ["Cherry", "Blueberry","Peach"]
 
 // b) Create the function that makes the test pass.
 
+// Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
+
+const filterLetter = (array, letter) => {
+const searchedWords = array.map(array.includes(letter));
+    for (let i = 0; i < array.length; i++) {
+        if (array.includes("a", "e")) {
+        searchedWords.push(array(letter))
+        }
+    }
```

Comment:
Again here, `.includes` expects 1 argument, but you are giving it 2 here.
The `.map` higher order function takes a callback function as an argument, which is not set up on line 118.
`array` on line 121 is not a function, so trying to perform a function call with it will output an error.